astria-conductor/src/main.rs

```diff
@@ -21,35 +23,15 @@ async fn main() -> Result<()> {
     // logs
     logger::initialize();
 
-    // cli args
-    let url_option = Arg::new("url")
-        .short('u')
-        .help("URL of the data layer server.")
-        .required(true);
-    let namespace_id_option = Arg::new("namespace_id")
-        .short('n')
-        .help("Namespace ID as a string; the hex encoding of a [u8; 8]")
-        .required(true);
-    let cli_app = Command::new("astria-conductor")
-        .version("0.1")
-        .about(
-            "A cli to read and write blocks from and to different sources. Uses the Actor model.",
-        )
-        .arg(url_option)
-        .arg(namespace_id_option);
-
-    let matches = cli_app.get_matches();
-    let base_url = matches.get_one::<String>("url").expect("url required");
-    let namespace_id = matches
-        .get_one::<String>("namespace_id")
-        .expect("namespace id required");
+    // parse cli args
+    let args = Cli::parse();
 
     // configuration
-    let conf = Conf::new(base_url.to_owned(), namespace_id.to_owned());
+    let conf = Conf::new(args.url, args.namespace_id, args.rpc_address);
```

Comment:
It doesn't look like I can use that in conjunction with cli args in an easy way. Do you think we should have config in a flat file instead of passing as cli args?

Reply:
You use a `figment::providers::Toml` for the conf file (or another file type), and then `figment::providers::Serialized` for the command line args. This requires that you `#[derive(serde::Serialize)]` on `Cli`.

Then you can `Figment::new().merge(Toml::file(your_file)).merge(Serialized::from(&args, "default")).extract()` to get your final config (with the command line args overriding your conf file args).

Reply:
Ah, I didn't read your question:

For deploying services it's really nice to not rely on command line args and instead use 1. a conf file, 2. env vars, and 3. cli args (with the latter always overriding the former).